### ISSUE TITLE
Streaming app can't connect to Redis in IPv6-only environment

### ISSUE THREAD

**Issue by ikuradon** [mastodon/mastodon#18715]
### Steps to reproduce the problem

1. Deploy Redis and mastodon on Amazon EKS with IPv6 option.
2. Start streaming Pod
3. View Pod log

### Expected behaviour

Running correctly

### Actual behaviour

Can't connect to Redis

### Specifications

Version: docker.io/tootsuite/mastodon:v3.5.3
(tootsuite/mastodon@sha256:bd1d81cd1948cb29e9fd640c902d377fc84fe2567589aca0c290b760d57b6a2b)
Log:
```
WARN Starting streaming API server master with 3 workers 
WARN Starting worker 1 
WARN Starting worker 2 
WARN Worker 1 now listening on :::4000 
WARN Worker 2 now listening on :::4000 
WARN Starting worker 3 
WARN Worker 3 now listening on :::4000 
ERR! [REDACTED] Error: Access token does not cover required scopes
ERR! [REDACTED] Error: Access token does not cover required scopes
ERR! Redis Client Error! Error: Socket closed unexpectedly
ERR! Redis Client Error!     at Socket.<anonymous> (/opt/mastodon/node_modules/@node-redis/client/dist/lib/client/socket.js:182:118)
ERR! Redis Client Error!     at Object.onceWrapper (node:events:646:26)
ERR! Redis Client Error!     at Socket.emit (node:events:526:28)
ERR! Redis Client Error!     at TCP.<anonymous> (node:net:687:12)
ERR! Redis Client Error!  SocketClosedUnexpectedlyError: Socket closed unexpectedly
ERR! Redis Client Error!     at Socket.<anonymous> (/opt/mastodon/node_modules/@node-redis/client/dist/lib/client/socket.js:182:118)
ERR! Redis Client Error!     at Object.onceWrapper (node:events:646:26)
ERR! Redis Client Error!     at Socket.emit (node:events:526:28)
ERR! Redis Client Error!     at TCP.<anonymous> (node:net:687:12)
ERR! Redis Client Error! Error: Socket closed unexpectedly
ERR! Redis Client Error!     at Socket.<anonymous> (/opt/mastodon/node_modules/@node-redis/client/dist/lib/client/socket.js:182:118)
ERR! Redis Client Error!     at Object.onceWrapper (node:events:646:26)
ERR! Redis Client Error!     at Socket.emit (node:events:526:28)
ERR! Redis Client Error!     at TCP.<anonymous> (node:net:687:12)
ERR! Redis Client Error!  SocketClosedUnexpectedlyError: Socket closed unexpectedly
ERR! Redis Client Error!     at Socket.<anonymous> (/opt/mastodon/node_modules/@node-redis/client/dist/lib/client/socket.js:182:118)
ERR! Redis Client Error!     at Object.onceWrapper (node:events:646:26)
ERR! Redis Client Error!     at Socket.emit (node:events:526:28)
ERR! Redis Client Error!     at TCP.<anonymous> (node:net:687:12)
ERR! Redis Client Error! Error: Socket closed unexpectedly
ERR! Redis Client Error!     at Socket.<anonymous> (/opt/mastodon/node_modules/@node-redis/client/dist/lib/client/socket.js:182:118)
ERR! Redis Client Error!     at Object.onceWrapper (node:events:646:26)
ERR! Redis Client Error!     at Socket.emit (node:events:526:28)
ERR! Redis Client Error!     at TCP.<anonymous> (node:net:687:12)
```

**Comment by ThisIsMissEm** [comment]
In the 4.2.0 release, we're moving from node-redis to ioredis, so it'll be worth you retesting against that release (currently 4.2.0-rc1 is out).

(I can't actually test this myself as I'm developing via Docker For Mac which apparently doesn't support ipv6.

**Comment by ThisIsMissEm** [comment]
Okay, this is fixed in 4.2.0, however, your `REDIS_URL` for streaming will need to specify it's using IPv6 by passing the `?family=6` parameter. See #27043
